### DISCUSSION TITLE
Why we need stop propagation here?

### DISCUSSION THREAD

**Discussion by RonNachmany** [adobe/react-spectrum#2688] [Q&A]
https://github.com/adobe/react-spectrum/blob/4312f197d683e75af962cf29fe7152c0377652b9/packages/%40react-aria/interactions/src/usePress.ts#L326

We are capturing the left mouse click events for closing dialogs and more..
Now, when users click on usePress elements the event is stopping, and I wonder if there is any reason for that.

**Comment by snowystinger** [comment]
This is the default for us. You can use capturing listeners (https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#parameters see capture) in order to get ahead of them for things like closing dialogs. Though if I'm reading that right, you're trying to close the dialog when someone clicks on a button outside of the dialog? If that's correct, you should really put up an underlay to prevent interactions with things outside the dialog. This will help mitigate the case you're talking about.
You are welcome to see how we handle interactions outside of dialogs or really all overlays which adds capturing listeners to the document. https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/interactions/src/useInteractOutside.ts

**Comment by RonNachmany** [comment]
Thanks for the fast and detailed response.
May you explain **why** you are stoping the propagation?
The closing dialog was just one example, we are counting on click events in several places and I don't thinks that button click should stop the propagation of click event.

**Comment by snowystinger** [comment]
Of course, sorry, I forgot to address the why in my previous message. This is probably the most relevant discussion explaining why we do it https://github.com/adobe/react-spectrum/issues/2100#issuecomment-878649270
